eth/ethconfig/config.go

```diff
@@ -232,13 +233,31 @@ type Config struct {
 
 // CreateConsensusEngine creates a consensus engine for the given chain configuration.
 func CreateConsensusEngine(stack *node.Node, chainConfig *params.ChainConfig, config *ethash.Config, notify []string, noverify bool, db ethdb.Database, l1Client sync_service.EthClient) consensus.Engine {
-	// If proof-of-authority is requested, set it up
+	// Case 1: Both SystemContract and Clique are defined: create an upgradable engine.
+	if chainConfig.SystemContract != nil && chainConfig.Clique != nil {
+		// Create the Clique engine.
+		cliqueEngine := clique.New(chainConfig.Clique, db)
+		// Create the SystemContract engine.
+		sysEngine := system_contract.New(context.Background(), chainConfig.SystemContract, l1Client)
+
+		// Determine the fork block at which the switch occurs.
+		var forkBlock *big.Int
```

Comment:
use the `isEuclid` function (pass into `NewUpgradableEngine`) to determine whether it is the Euclid fork. https://github.com/scroll-tech/go-ethereum/pull/1104/files#diff-10ff00a15ef58da7485fedeca3906c73cb236fde4b143a9d61601c63cbf1ffe8R899